### PR TITLE
Add missing header for CMake, add more examples, add generic type support for NVSHMEM back-end

### PR DESCRIPTION
**Pull request by janciesko**

### CODE REVIEW

File: examples/vectorshift/single-node/vectorshift.cpp [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    Kokkos::deep_copy(a_h, b);
+    //Excersize: Update error check to check if value "1" has been shifter
+    //Note: it resides porentially on a different PE
+    assert(a_h((NUM_SHIFTS*OFFSET%n)-1)==1);
+  }
+  Kokkos::finalize();
+
+  // Excersize: Uncomment networking finalization below
+  /*
+  #ifdef KOKKOS_ENABLE_SHMEMSPACE
```
Comment (jjwilke) [inline examples/vectorshift/single-node/vectorshift.cpp:132]: Why is this commented here?

File: unit_tests/Test_LocalDeepCopy.cpp [hunk 2]
```diff
@@ -123,8 +124,10 @@ void test_localdeepcopy(int i1,
                 v_R(my_rank,j) = 0x123;
               });
       
+      //Run 1 Team only.
       Kokkos::single(Kokkos::PerTeam(team), [&]() {
-        Kokkos::Experimental::local_deep_copy(team, v_R_cpy, v_R);
+        Kokkos::Experimental::local_deep_copy(v_R_cpy, v_R);
+        //Kokkos::Experimental::local_deep_copy(team, v_R_cpy, v_R); //Currently not working. TBD
```
Comment (jjwilke) [inline unit_tests/Test_LocalDeepCopy.cpp:130]: What isn't working? I'm a bit uncomfortable committing this.

File: examples/vectorshift/multi-node/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+add_executable(shift_multi vectorshift.cpp)
+target_link_libraries(shift_multi PRIVATE Kokkos::kokkosremote)
+target_include_directories(shift_multi PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
```
Comment (jjwilke) [inline examples/vectorshift/multi-node/CMakeLists.txt:3]: Do we actually need to include the current source dir?

File: unit_tests/Test_LocalDeepCopy.cpp (hunk 2 above)
Comment (janciesko) [reply unit_tests/Test_LocalDeepCopy.cpp:130]: The overload of local_deep_copy that takes the team parameter performs a local deep copy using a parallel_for over a flattened iteration space and reconstructs the indexes from a flat index i, shown [here](https://github.com/kokkos/kokkos/blob/03ded4e816da5927b946065b005ebfdf38e5a438/core/src/Kokkos_CopyViews.hpp#L1676). The reconstruction is incorrect for our case. Making this work requires to specialize local_deep_copy to a remote space.

File: examples/vectorshift/single-node/vectorshift.cpp (hunk 1 above)
Comment (janciesko) [reply examples/vectorshift/single-node/vectorshift.cpp:132]: Thanks for pointing this out, this is an artifact from the exercise. Correcting.

**Comment by janciesko** [comment]
Closing PR as this would introduce the calls nvshmem_put{4,8,32,64,128} in the nvshmem backend which currently generates a rma completion error. Requires further testing.

mlx5: lassen33: got completion with error:
00000000 00000000 00000000 00000000
00000000 00000000 00000000 00000000
00000003 00000000 00000000 00000000
00000000 9d003304 1003b179 000358d2
src/comm/transports/ibrc/ibrc.cpp:938: non-zero status: 4 ibv_poll_cq failed, status: 4
src/comm/transports/ibrc/ibrc.cpp:993: non-zero status: 7 progress_send failed, outstanding_count: 1023
src/comm/transports/ibrc/ibrc.cpp:1017: non-zero status: 7 check_poll failed
src/comm/proxy/proxy.cu:process_channel_dma:294: aborting due to error in process_channel_dma

**Pull request closed without merging**